src/GitHub.Api/Git/RepositoryManager.cs

```diff
@@ -173,20 +173,20 @@ public int WaitForEvents()
 
         public ITask CommitAllFiles(string message, string body)
         {
-            var add = GitClient.AddAll();
-            add.OnStart += t => IsBusy = true;
-            return add
-                .Then(GitClient.Commit(message, body))
-                .Finally(() => IsBusy = false);
+            var task = GitClient.AddAll()
+                .Then(GitClient.Commit(message, body));
+
+            HookupHandlers(task, true);
```

Comment:
Hmmmm this will only hook up handlers for the `Commit` task, not the `AddAll` task (which will also fire events). Is that what we want? If we want to handle the whole chain of tasks in one go, we can hook up to the `Start` of the `AddAll` task and do a `Finally` on the `Commit` one (so that if the `AddAll` fails we can still reset the busy state and restart the watcher - if the `AddAll` fails the `Commit` will never start so handlers in `HookupHandlers` for it will never run.